afs/afs-mapdb-storage/src/main/java/com/powsybl/afs/mapdb/storage/MapDbAppStorage.java

```diff
@@ -301,31 +324,41 @@ public boolean isWritable(String nodeId) {
         return true;
     }
 
+    @Override
+    public boolean isConsistent(String nodeId) {
+        checkNodeId(nodeId);
+        NodeInfo nodeInfo = getNodeInfo(nodeId);
+        return nodeInfo.isConsistent();
+    }
+
     @Override
     public List<NodeInfo> getChildNodes(String nodeId) {
         UUID nodeUuid = checkNodeId(nodeId);
+        checkConsistent(nodeUuid);
         List<UUID> childNodes = childNodesMap.get(nodeUuid);
         if (childNodes == null) {
             throw createNodeNotFoundException(nodeUuid);
         }
-        return childNodes.stream().map(this::getNodeInfo).collect(Collectors.toList());
+        return childNodes.stream().map(this::getNodeInfo).filter(NodeInfo::isConsistent).collect(Collectors.toList());
```

Comment:
We must make clear, somewhere in the API documentation, that those methods must filter out "inconsistent" nodes. I mean for the people who want to implement that interface, not so much for users.

Reply:
@geofjamg Should we document in the javadoc ?

Reply:
After discussing that with @geofjamg  : doc will be added to the API javaDoc